src/main/java/com/minecolonies/api/tools/registry/ToolTypeEntry.java

```diff
@@ -0,0 +1,216 @@
+package com.minecolonies.api.tools.registry;
+
+import com.minecolonies.api.MinecoloniesAPIProxy;
+import com.minecolonies.api.tools.ModToolTypes;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * An entry in the ToolType registry that defines the types of
+ * tools within the colony.
+ */
+public final class ToolTypeEntry
+{
+    /**
+     * The registry identifier for this tool type.
+     */
+    private final ResourceLocation registryName;
+
+    /**
+     * The name of the tool type.
+     */
+    private final String name;
+
+    /**
+     * The component for the human readable name.
+     */
+    private final Component displayName;
+
+    /**
+     * Predicate to determine whether a given ItemStack
+     * can act as this tool type.
+     */
+    private final Predicate<ItemStack> isTool;
+
+    /**
+     * A function to return the integer item level of a
+     * given ItemStack.
+     */
+    private final Function<ItemStack, Integer> itemLevel;
+
+    /**
+     * Constructor.
+     *
+     * @param name        The name of the tool type
+     * @param displayName The human readable name of the tool type
+     * @param isTool      A predicate for determining if an itemstack is the tool type
+     * @param itemLevel   A function to return the item level of an item stack
+     */
+    private ToolTypeEntry(final String name, final Component displayName, final Predicate<ItemStack> isTool, final Function<ItemStack, Integer> itemLevel, final ResourceLocation registryName)
```

Comment:
Order of constructor is not identical to field order, would expect registryName in front (also missing javadoc for registryName)

Reply:
Still applies